sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -253,6 +254,18 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       (1, 1))
   }
 
+  test("MapPartitions can process unsafe rows") {
```

Comment:
I am not sure we need this test.  Its pretty brittle if we make more things operate on unsafe rows (and we should).